### PR TITLE
Ceph: Add an rbd hang up problem in the common issues

### LINKED ISSUE #3132: Very high CPU usage on Ceph OSDs (v1.0, v1.1)
I am not sure where the problem is but I am seeing very high CPU usage since I started using v1.0.0. With three small clusters load average skyrockets to the 10s quite quickly making the nodes unusable. This happens while copying quite a bit of data to a volume mapped on the host bypassing k8s (to restore data from an existing non-k8s server). Nothing else is happening with the clusters at all. I am using low specs servers (2 cores, 8 GB of RAM) but I didn't see any of these high load issues with 0.9.3 on same-specs servers. 
Has something changed about Ceph or else that might explain this? I've also tried with two providers, Hetzner Cloud and UpCloud. Same issue when actually using a volume.

Is it just me or is it happening to others as well? Thanks!

### PR DESCRIPTION
**Pull request by satoru-takeuchi**
Signed-off-by: Satoru Takeuchi <satoru.takeuchi@gmail.com>

**Description of your changes:**

Add the description about the following issue to the common issues.

https://github.com/rook/rook/issues/3132

**Which issue is resolved by this Pull Request:**

Resolves #3132 

**Checklist:**

- [x] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/master/development-flow.html#submitting-a-pull-request)
- [x] Documentation has been updated, if necessary.
- [x] Unit tests have been added, if necessary.
- [x] Integration tests have been added, if necessary.
- [x] Pending release notes updated with breaking and/or notable changes, if necessary.
- [x] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [x] Code generation (`make codegen`) has been run to update object specifications, if necessary.
- [x] Comments have been added or updated based on the standards set in [CONTRIBUTING.md](https://github.com/rook/rook/blob/master/CONTRIBUTING.md#comments)
- [x] Add the flag for skipping the CI if this PR does not require a build. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for more details.

[skip ci]

### CODE REVIEW

**Comment by galexrt** [comment]
You have forgotten to add `[skip ci]` to your doc only PR, as it is written in the checklist ;-)

I have added and restarted the CI.

File: Documentation/ceph-common-issues.md [hunk 1]
```diff
@@ -684,3 +685,47 @@ Ceph CSI in its 1.2 version (with Rook 1.1) does not support the Erasure coded p
...
+For more detail, please refer to [the corresponding issue comment](https://github.com/rook/rook/issues/3132#issuecomment-580508760).
+
+### Solution
+
+This problem will be solve by the following two fixes.
+
+* Linux kernel: A minor feature that is introduced by [this commit](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=8d19f1c8e1937baf74e1962aae9f90fa3aeab463). It will be included in Linux v5.6.
+* Ceph: A fix that uses the above-mentioned kernel's feature. The Ceph community will probably discuss this fix after releasing Linux v5.6.
+
+In most cases, you can bypass this problem by using ext4 (or any other filesystems) rather than XFS. Although this problem might happen with other filesystems, it's possibility is very low.
```
Comment (leseb) [inline Documentation/ceph-common-issues.md]: Can we specify the related csi option of the storage class for this?

Comment (idryomov) [reply Documentation/ceph-common-issues.md]: Should be `csi.storage.k8s.io/fstype`, but rook and ceph-csi default to ext4.

Comment (satoru-takeuchi) [reply Documentation/ceph-common-issues.md]: Thanks,  I mentioned csi.storage.k8s.io/fstype in the new PR.

File: Documentation/ceph-common-issues.md [hunk 2]
```diff
@@ -684,3 +685,47 @@ Ceph CSI in its 1.2 version (with Rook 1.1) does not support the Erasure coded p
...
+* I/O to an RBD device: It allocates some memory in the kernel.
+* The kernel's memory reclaiming logic: It's kicked typically under low memory environment and incurs I/O to that device.
+
+The probability of this problem becomes high if the following conditions are satisfied.
+
+* XFS is created on the problematic RBD device.
+* This device and the corresponding OSDs are co-located.
+* There is high I/O pressure to this device.
+* There is lack of free memory in the worker node.
+* The Rook/Ceph cluster is under upgrading.
```
[the remark this answers is not in the document]
Comment (satoru-takeuchi) [reply Documentation/ceph-common-issues.md]: Thank you for letting me know my misunderstanding, fixed. In addition, I also fixed the comment in the original issue based on your comment.

File: Documentation/ceph-common-issues.md [hunk 3]
```diff
@@ -684,3 +685,47 @@ Ceph CSI in its 1.2 version (with Rook 1.1) does not support the Erasure coded p
...
+...
+[51717.039319] INFO: task kworker/2:1:5938 blocked for more than 120 seconds.
+[51717.039361]       Not tainted 4.15.0-72-generic #81-Ubuntu
+[51717.039388] "echo 0 > /proc/sys/kernel/hung_task_timeout_secs" disables this message.
+...
+```
+
+It's so-called `hung_task` problem and means that there is a deadlock in the kernel. Here the following two logics cause the deadlock.
+
+* I/O to an RBD device: It allocates some memory in the kernel.
```
[the remark this answers is not in the document]
Comment (satoru-takeuchi) [reply Documentation/ceph-common-issues.md]: Now I understood correctly, fixed.

**Comment by satoru-takeuchi** [comment]
@leseb @idryomov I fixed this PR based on your comments. Could you take a look at it again?

**Pull request merged**